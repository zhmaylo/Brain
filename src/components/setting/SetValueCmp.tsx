import React from 'react';
import { StyleSheet, Text, TextInput, View } from "react-native";
import { getValueStore, setValueStore } from '../../api/setting/setting';
import { use_State } from '../../api/setting/use_State';


export const SetValueCmp = ({ title, keyStore, valueDef, onChangeValue, secure=false }) => {
    let { value, setValue } = use_State('');
    getValueStore(keyStore, valueDef).then((value: string) => { setValue(value) });

    return (
        <View style={styles.itemValue}>
            <Text style={styles.titleValue}> {title}  </Text>
            <TextInput
                style={styles.inputValue}
                defaultValue={value}
                secureTextEntry={secure}
                onChangeText={(value) => {
                    setValueStore(keyStore, value);
                    onChangeValue(value);
                }}
            />
        </View>
    )
};

const styles = StyleSheet.create({
    itemValue: {
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: 'white',
        padding: 5,
        width: 400,
    },
    titleValue: {
        fontSize: 18,
        fontWeight: "600",
        width: 100,
    },
    inputValue: {
        borderColor: "#f0f0f0",
        color: 'black',
        borderWidth: 2,
        height: 50,
        width: 250,
        backgroundColor: 'white',
        textAlign: 'center',
        textAlignVertical: 'center',

        fontSize: 18,
        fontWeight: "600",
    }
})

            