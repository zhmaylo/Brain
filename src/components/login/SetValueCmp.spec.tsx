import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);
import { SetValueCmp } from './SetValueCmp';
import renderer from 'react-test-renderer';
// import { useState } from 'react';

jest.mock('../../api/login/use_State');
test('renders correctly', () => {

    let title: any = 'test title';
    let keyStore: any = 'key';
    let valueDef: any = 'valueDef';
    let onChangeValue: any;
    let secure = true;

    const tree = renderer.create(
        SetValueCmp({ title, keyStore, onChangeValue, secure })).toJSON();

        expect(tree).toMatchSnapshot();
});
