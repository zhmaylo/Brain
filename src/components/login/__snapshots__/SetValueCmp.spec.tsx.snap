// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "padding": 5,
      "width": 400,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "600",
        "width": 100,
      }
    }
  >
     
    test title
      
  </Text>
  <TextInput
    allowFontScaling={true}
    defaultValue="1"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#f0f0f0",
        "borderWidth": 2,
        "color": "black",
        "fontSize": 18,
        "fontWeight": "600",
        "height": 50,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 250,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
